pandas/core/reshape/melt.py

```diff
@@ -40,6 +41,16 @@ def melt(
     else:
         cols = list(frame.columns)
 
+    if value_name in frame.columns:
+        warnings.warn(
+            "This dataframe has a column name that matches the value column "
```

Comment:
put `value` in single quotes as this is an argument that the user passed. (and this should be `value_name`)